src/main.cpp

```diff
@@ -127,12 +166,68 @@ Mat RenderFrame(void)
 
 		// --- PUT YOUR CODE HERE ---
 		// Apply transforms here 
-		Mat rotationAroundTheSun = Mat::eye(4, 4, CV_32FC1);
+		//get the Earth's pivot 
+		Vec3f pivot = earth.getPivot();
+		//degrees for rotation of Earth around its axis 
+		float degreesperframeer = 1.0f / nFrames;
+		Mat rotationAroundTheSun = transform.translate(pivot).rotate(Vec3f(0, 1, 0), degreesperframeer).translate(-pivot).get();
 		earth.transform(rotationAroundTheSun * earthTransform);
+		moon.setPivot(earth.getPivot());
 		moon.transform(rotationAroundTheSun * moonTransform);
 
 		// --- PUT YOUR CODE HERE ---
 		// Apply camera animation here
+
+		//target animation
+		float steptar = frame / nFrames;
+		//get the linear spline 
+		Vec3f targetspl = tarframe1 - tarframe0;
+		//get the norm
+		Vec3f targetnorm = norm(targetspl);
+		targetspl = normalize(targetspl);
+		Vec3f newtar = tarframe0 + (targetspl * (targetnorm * steptar));
+		cam3->setTarget(newtar);
+
+		//origin animation
+		//put loops to check for the conditions
+		if (frame == 0) {
+			Vec3f point=posframe0;
+			cam3->setPosition(point);
+		}
+		else if (frame > 0 && frame < (nFrames / 2)) {
+			//get the linear spline 
+			Vec3f orspl = posframe1 - posframe0;
+			Vec3f orgnorm = norm(orspl);
+			orspl = normalize(orspl);
+			float steppos = frame / (nFrames / 2);
```

Comment:
Stepps all the time is equal to 0. You had to use 
`float steppos = static_cast<float>(frame) / (nFrames / 2);` instead